game.js

```diff
@@ -18,74 +19,44 @@ class GameRoom extends Room {
       const players = this.state.players
       const player = players[client.sessionId]
 
-      switch (direction) {
-        case 'up':
-
-          if (player.y + 1 < mapY) {
-            const playerCollidedId = isCollide(player.x, player.y + 1, players)
-            if (playerCollidedId) {
-              players[playerCollidedId].tagPlayer(player)
-            } else {
-              player.y += 1
-            }
-          }
-          break
-
-        case 'down':
-
-          if (player.y - 1 >= 0) {
-            const playerCollidedId = isCollide(player.x, player.y - 1, players)
-            if (playerCollidedId) {
-              players[playerCollidedId].tagPlayer(player)
-            } else {
-              player.y -= 1
-            }
-          }
-          break
-
-        case 'left':
-
-          if (player.x - 1 >= 0) {
-            const playerCollidedId = isCollide(player.x - 1, player.y, players)
-            if (playerCollidedId) {
-              players[playerCollidedId].tagPlayer(player)
-            } else {
-              player.x -= 1
-            }
-          }
-          break
-
-        case 'right':
-
-          if (player.x + 1 < mapX) {
-            const playerCollidedId = isCollide(player.x + 1, player.y, players)
-            if (playerCollidedId) {
-              players[playerCollidedId].tagPlayer(player)
-            } else {
-              player.x += 1
-            }
-          }
-          break
+      const moveMap = {
+        up: [0, 1],
+        down: [0, -1],
+        left: [-1, 0],
+        right: [1, 0]
       }
+
+      const moveOffset = moveMap[direction]
+      if (!moveOffset) return
+      const newX = player.x + moveOffset[0]
+      const newY = player.y + moveOffset[1]
+      if (!inRange(newX, 0, mapX - 1) || !inRange(newY, 0, mapY - 1)) return
+
+      const collidedPlayer = isCellOccupied(newX, newY, players)
+      if (collidedPlayer) return collidedPlayer.tagPlayer(player)
+
+      player.x = newX
+      player.y = newY
     })
   }
 
   onJoin (client, options) {
     console.log('client ', client.sessionId, ' joined')
 
     const playerName = options?.name
-    if (!playerName || !validatePlayerName(playerName)) {
+    if (!playerName || !Player.validateName(playerName)) {
+      console.log('client ' + client.sessionId + ' hasn\'t provided a valid name')
```

Comment:
To keep consistency with the rest of the code (read up), this could be:
```javascript
console.log('client ', client.sessionId, ' hasn\'t provided a valid name')
```